crates/cairo-1-hint-processor/src/hint_processor.rs

```diff
@@ -69,6 +72,26 @@ fn res_operand_get_val(
     }
 }
 
+/// Fetches the value of `res_operand` from the vm.
+fn get_val(vm: &VirtualMachine, res_operand: &ResOperand) -> Result<Felt252, VirtualMachineError> {
+    match res_operand {
+        ResOperand::Deref(cell) => get_cell_val(vm, cell),
+        ResOperand::DoubleDeref(cell, offset) => get_double_deref_val(vm, cell, &(*offset).into()),
+        ResOperand::Immediate(x) => Ok(Felt252::from(x.value.clone())),
+        ResOperand::BinOp(op) => {
+            let a = get_cell_val(vm, &op.a)?;
+            let b = match &op.b {
+                DerefOrImmediate::Deref(cell) => get_cell_val(vm, cell)?,
+                DerefOrImmediate::Immediate(x) => Felt252::from(x.value.clone()),
+            };
+            match op.op {
+                Operation::Add => Ok(a + b),
+                Operation::Mul => Ok(a * b),
+            }
+        }
+    }
+}
+
```

Comment:
currently in the branch 0.11 has been merge a function that does exactly this but has a different name `res_operand_get_val.` i would recommend to do a pull from that branch and merge it to this one.